### DISCUSSION TITLE
Tidy up conditions for usage of `tokenTypeToLexicalStateMap`?

### DISCUSSION THREAD

**Discussion by vsajip** [javacc21/javacc21#186] [Q&A] [answered]
This declaration is unconditional:
https://github.com/javacc21/javacc21/blob/ba4f192a8f7b07af6981530a03074e1004baec71/src/ftl/java/Lexer.java.ftl#L151-L152
However, its usages are conditional, but for different conditions:
https://github.com/javacc21/javacc21/blob/ba4f192a8f7b07af6981530a03074e1004baec71/src/ftl/java/Lexer.java.ftl#L393-L399
and
https://github.com/javacc21/javacc21/blob/ba4f192a8f7b07af6981530a03074e1004baec71/src/ftl/java/Lexer.java.ftl#L452-L461
It certainly seems possible for `multipleLexicalStates` (which is equivalent to `lexerData.lexicalStates.size() > 1`) to be true and `lexerData.hasLexicalStateTransitions` to be false.

**Comment by revusky** [comment, marked as answer]
> It certainly seems possible for multipleLexicalStates (which is equivalent to lexerData.lexicalStates.size() > 1) to be true and lexerData.hasLexicalStateTransitions to be false.

Yes, that's certainly true and in those cases, there really isn't any need to generate the `doLexicalStateSwitch` method, since it would always just do nothing and return false, since the `tokenTypeToLexicalStateMap` is just empty.

So, yeah, the code was definitely a bit sloppy and I'm not sure why it was like that, but now I fixed it up. Thanks.
